Add IQS043 rule (#2407)
home-assistant/developers.home-assistant: docs/core/integration-quality-scale/rules/iqs043.md

---
title: "IQS043 - The documentation provides removal instructions"
---

IQS043 is as follows:

> The documentation provides removal instructions

Reviewer: ```suggestion
The user documentation provides removal instructions
```

## Reasoning

Removing a device or service from Home Assistant isn't always straightforward.

> The documentation should provide clear instructions on how to remove it, even if it's as simple as "go the integrations page and click delete".

Reviewer 2: I suggest we remove this example as it's not complete. You can't click delete on the integrations page. To make it complete it would just be a repetition of the actual example below.
```suggestion
The documentation should provide clear instructions on how to remove the device or service.
```

## Example implementation

```markdown

> To remove the integration, go to the integrations page and click delete.

Reviewer: ```suggestion
To remove the integration, go to {% my integrations title="**Settings** > **Devices & services**" %} and select the integration card. Then, select the three dots {% icon "mdi:dots-vertical" %} menu and select **Delete**.
```

> Make sure that after you delete the integration, you go to the app of the manufacturer and remove the Home Assistant integration from there as well.

Reviewer: ```suggestion
After deleting the integration, go to the app of the manufacturer (if any) and remove the Home Assistant integration from there as well.
```

> ```

Reviewer: should we also add a note about removing any stored credentials? Or is that a separate topic?

Author: Oh, that makes sense for integrations that have application credentials like spotify and withings. I think that is just another example of how it will be implemented. I don't think we necessarily have to add it to the example for now
